modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -76,4 +81,10 @@ public String toString() {
         searchToString(b);
         return b.toString();
     }
+
+    @Override
+    public List<? extends IndicesRequest> subRequests() {
+        assert getSearchRequest() != null;
+        return Collections.singletonList(getSearchRequest());
```

Comment:
Sorry for the back and forth, I had to do some more digging. CompositeIndicesRequest is not really about the type of operation (read or write etc.) but more about each set of indices and their corresponding indices options that are used to resolve them. Think of _msearch for instance, each inner search has its own indices, which get resolved based on each individual indices options.

That said, I am not sure that dele by query should be seen as a composite request, as the indices get resolved once when doing the search. Then it should simply implement IndicesRequest and return the set of indices that the inner search request points to.